NOTICK: Mention platform version change in release notes

lockathan opened the pull request:
Update the release notes to mention the platform version change.

lockathan committed:
NOTICK: Mention platform version change in release notes

Update the release notes to mention the platform version change.

lockathan commented:
I will also be raising a separate PR to update the versioning.rst file. This is because that change needs to go into 4.1 and be forward ported.

gendal commented on docs/source/release-notes.rst:
```diff
@@ -88,6 +88,13 @@ There have been several security upgrades, including changes to the Corda webser
+~~~~~~~~~~~~~~~~~~~~~~~
+
+As some API calls were added to Corda to support Accounts, the platform version of Corda 4.3 needed to be changed from 4 to 5. This allows CorDapps that use Accounts to set their minimum platform version to 5, ensuring that they only operate on nodes that support the required features. CorDapps written for older versions of Corda are unaffected and will function without modification. Corda nodes  ...
+
```
Maybe add: "Since the APIs added do not affect the wire protocol or have other zone-level implications, applications can take advantage of these new platform version 5 features even if the Corda 4.3 is running on a network whose minimum platform version is 4."

lockathan committed:
NOTICK: Mention platform version change in release notes

Added mention of preserved wire stability and compatibility across different versions.

Commit:
Typo fixes

lockathan committed:
Corrected typo in release notes.

lockathan committed:
Merge branch 'jlocke/4.3_release_notes_update' of github.com:corda/corda into jlocke/4.3_release_notes_update

fenryka commented on docs/source/release-notes.rst:
```diff
@@ -88,14 +88,21 @@ There have been several security upgrades, including changes to the Corda webser
+Platform version change
+~~~~~~~~~~~~~~~~~~~~~~~
+
+As some API calls were added to Corda to support Accounts, the platform version of Corda 4.3 needed to be changed from 4 to 5. This allows CorDapps that use Accounts to set their minimum platform version to 5, ensuring that they only operate on nodes that support the required features. CorDapps written for older versions of Corda are unaffected and will function without modification. Corda nodes  ...
```
I must admit this is somewhat confusing to read, maybe a table / matrix of what will and won’t work would be a bit clearer

Additionally, I would phrase the start slightly differently.

“Given the addition of a new API to support the Accounts feature, the platform version of 4.3 has been bumped up to 5. As per our spec this is to prevent apps that utilise it being deployed onto nodes unable to host them.”

The current phrasing is a bit too matter of fact

lockathan committed:
Reworded change according to feedback - greatly simplified it.

fenryka commented on docs/source/release-notes.rst:
```diff
@@ -88,14 +88,21 @@ There have been several security upgrades, including changes to the Corda webser
+Platform version change
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the addition of a new API to support the Accounts feature, the platform version of Corda 4.3 has been bumped up from 4 to 5. This is to prevent CorDapps that use it being deployed onto nodes unable to host them. Since the APIs added do not affect the wire protocol or have other zone-level implications, applications can take advantage of these new platform version 5 features even if the Cord ...
```
My only remaining comment is can we make it clearer that v5 apps *will* run on a mpv 4 zone, perhaps as a ..important style thing

fenryka reviewed (changes requested):
just one more thing I think and it's good

lockathan committed:
Slight rewording

fenryka reviewed (approved):
LGTM
